hydra-plutus/src/Hydra/Contract/Head.hs

```diff
@@ -126,21 +129,37 @@ checkCollectCom commitAddress (_, parties) context@ScriptContext{scriptContextTx
     && everyoneHasCommitted
  where
   everyoneHasCommitted =
-    -- TODO: This check is lame, what we want is to check also the PTs are legit, which means
-    -- verifying the CurrencySymbol match the Head.
-    collectedPTs == length parties
+    length collectedPTs == length parties
+
+  threadToken =
+    context
+      & findOwnInput
+      & maybe mempty (txOutValue . txInInfoResolved)
+      & symbols
+      & filter (/= adaSymbol)
+      & \case
+        [headPolicyId] -> headPolicyId
+        _ -> traceError "malformed thread token, expected single asset"
```

Comment:
Nice code. We should use consistent names though.

Should: As it seems not to be the thread token we retrieve, but this head's currency symbol, please rename `threadToken`.

Could: More minor, the term `headPolicyId` is not consistent with what plutus is calling it (ref `symbols` and `adaSymbol`). So maybe we should also be consistent in that regards?

Reply:
> As it seems not to be the thread token we retrieve, but this head's currency symbol

Yup, I told myself the same thing yesterday when I self reviewed the code. It's not a thread token indeed.